File: docs/alphacam-ecosystem/alphacam-provided-examples/API/CSharp.Net/RunAcam (CSharp)/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

// Alphacam
using AlphaCAMRouter;
using AlphaCAMMill;

namespace RunAcam__CSharp_
{
    public partial class Form1 : Form
    {
        // Acam objects
        AlphaCAMRouter.App AcamRouter;
        AlphaCAMMill.App AcamMill;

        // Global
        bool IsRouter;

        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Initialize Alphacam Router
            AcamRouter = new AlphaCAMRouter.App();
            IsRouter = true;

            textBox1.Text = AcamRouter.AlphacamVersion.String;
        }

        private void button4_Click(object sender, EventArgs e)
        {
            // Initialize Alphacam Router
            AcamMill = new AlphaCAMMill.App();
            IsRouter = false;

            textBox2.Text = AcamMill.AlphacamVersion.String;
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if (IsRouter && AcamRouter != null)
            {
                AlphaCAMRouter.Drawing Drw = AcamRouter.ActiveDrawing;

                Drw.CreateRectangle(0, 0, 100, 100);
            }
            else if (!IsRouter && AcamMill != null)
            {
                AlphaCAMMill.Drawing Drw = AcamMill.ActiveDrawing;

                Drw.CreateRectangle(0, 0, 100, 100);
            }
        }

        private void button3_Click(object sender, EventArgs e)
        {
            if (IsRouter && AcamRouter != null)
            {
                AcamRouter.Frame.RunCommand(AlphaCAMRouter.AcamCommand.acamCmdMACHINE_SELECT_TOOL);
            }
            else if (!IsRouter && AcamMill != null)
            {
                AcamMill.Frame.RunCommand(AlphaCAMMill.AcamCommand.acamCmdMACHINE_SELECT_TOOL);
            }
        }


    }
}

File: docs/alphacam-ecosystem/alphacam-provided-examples/API/VB.Net/RunAcam/RunAcam/Form1.vb
Imports Microsoft.Win32 ' for RegistryKey

' To add a reference to the Alphacam type library:
' Right-click on the Project name in the Solution Explorer and pick Properties
' Click the References tab and then the Add button
' Select the Alphacam module required
' In the Imported namespaces box select the added library 

Public Class Form1
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        ' Uncomment the next line to control how Alphacam starts up
        'SetStartUpOptions()
        Dim Acam As AlphaCAMRouter.App
		
		' create instance of Alphacam Advanced Router
		Acam = CreateObject("aroutaps.application")
		
        If Acam.ApiVersion >= 20070810 Then
            'This is V7.5 or later, so we can get the version and show it in the form
            TextBox1.Text = Acam.AlphacamVersion.String
            TextBox2.Text = Acam.ProgramLevel
        End If
        Dim Drw As Drawing
        Drw = Acam.ActiveDrawing
        If Acam.ProgramLevel <> AcamLevel.acamLevelPROUTAPS Then
            Drw.CreateText2("From VB.Net", 0, 0, 10)
        End If
        Drw.CreateRectangle(0, 0, 150, 100)
        Drw.ZoomAll()
        Drw.SaveAsEx(Acam.LicomdirPath + "licomdir\123.ard", AcamSaveAsVersion.acamSaveAsV7)
        'Acam.Quit()
        ' The following 4 lines make .Net release the Alphacam object, otherwise Alphacam stays in memory (until this program exits)
        Acam = Nothing
        System.GC.Collect()
        System.GC.WaitForPendingFinalizers()
        System.GC.Collect()
    End Sub

    Private Sub SetStartUpOptions()
        Dim Rk As RegistryKey
        Rk = Registry.CurrentUser.CreateSubKey("SOFTWARE\LicomSystems\Acam\COM", RegistryKeyPermissionCheck.ReadWriteSubTree)
        Rk.SetValue("ProgramLevel", AlphaCAMRouter.AcamLevel.acamLevelSTANDARD, RegistryValueKind.DWord)
    End Sub
End Class
